src/libs/DistanceRequestUtils.ts

```diff
@@ -146,6 +245,52 @@ function getDistanceRequestAmount(distance: number, unit: Unit, rate: number): n
     return Math.round(roundedDistance * rate);
 }
 
-export default {getDefaultMileageRate, getDistanceMerchant, getDistanceRequestAmount};
+/**
+ * Extracts the distance from a merchant string.
+ *
+ * @param merchant - The merchant string containing the distance.
+ * @returns The distance extracted from the merchant string.
+ */
+function getDistanceFromMerchant(merchant: string | undefined, unit: Unit): number {
+    if (!merchant) {
+        return 0;
+    }
+
+    const distance = Number(merchant.split(' ')[0]);
+    if (!distance) {
```

Comment:
NAB: It would also be a good idea to validate that the merchant has the expected format and return 0 if it does not. The regex we're using on the backend is found in a comment I posted [here](https://github.com/Expensify/App/issues/36987#issuecomment-2051952580). Let's do this in a follow up.

Reply:
@neil-marcellini FYI: I needed to change this regex a bit, `/^[0-9.]+ \w+ @ (-|-\()?(\p{Sc}|\p{L}|\w){1,3} ?[0-9.]+\)? \/ \w+$/u` - added an "u" flag to catch unicode and this one: `\p{L}` bc the regex didn't work when we had PLN currency - "zł"

Reply:
Ok great thanks. If you would please provide some examples where it breaks with the old regex, then I can get the backend fixed too so that the regexes match.